department-of-veterans-affairs/vets-website
# CST | Adding mock data for stem claims

## Original Issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/44034

## URL of change
[/track-claims/your-claims](http://localhost:3001/track-claims/your-claims)

## Background
The CST displays the current status of multiple different types of claims. The first is benefits claims, the second is appeals, and the third is education benefits claims (I'm not entirely sure why this is separate). Currently we are only mocking the first two types locally. This PR is to add mocking for education benefits claims.

## Steps to test
* Login with a staging user and navigate to: [/track-claims/your-claims](http://localhost:3001/track-claims/your-claims)
* There should be a section labeled 'Your claims and appeals'
* You should see a list of cards
* 3 of those cards should be labeled 'Edith Nourse Rogers STEM Scholarship application updated on...'

## Code changes
### Core
* Added JSON file containing list of fake STEM claims
* Update mockApi to return fake list of STEM claims
* Updated `getStemClaims` to use the mockApi locally

### Miscellaneous
* Added a extra flag for mocking that can be used to turn off local mocking manually
* Refactored some of the arrow functions to use more succinct syntax
* Refactored the step that adds attributes to each mock STEM claim to a separate function
* Moved claims JSON files into their own folder

## How was your code tested
Tests still pass

## Acceptance criteria
- [x] STEM claims are mocked when running vets-website locally
